services/vm-watcher/vm_watcher_service_test.go

```diff
@@ -356,6 +357,178 @@ var _ = Describe(
 					})))
 				})
 
+				When("a Zone that is being deleted", func() {
+					When("the zone has the zone controller finalizer", func() {
+						Specify("a reconcile request should be received", func() {
+							By("verify that the zone has the finalizer added", func() {
+								Eventually(func(g Gomega) {
+									zoneList := topologyv1.ZoneList{}
+									err := vcSimCtx.Client.List(vcSimCtx, &zoneList, ctrlclient.InNamespace(vcSimCtx.NSInfo.Namespace))
+									g.Expect(err).NotTo(HaveOccurred())
+									g.Expect(zoneList.Items).ToNot(BeEmpty())
+									for _, zone := range zoneList.Items {
+										g.Expect(zone.Finalizers).To(ContainElement(zonectrl.Finalizer))
+									}
+								})
+							})
+
+							By("a reconcile request should be received", func() {
+								chanSource := cource.FromContext(ctx, "VirtualMachine")
+								var e event.GenericEvent
+								Eventually(chanSource).Should(Receive(&e, Equal(event.GenericEvent{
+									Object: &vmopv1.VirtualMachine{
+										ObjectMeta: metav1.ObjectMeta{
+											Namespace: vcSimCtx.NSInfo.Namespace,
+											Name:      vmName,
+										},
+									},
+								})))
```

Comment:
Ditto.

Reply:
This is `Eventually().Should(Receive())`